apps/client/src/common/hooks/useEventAction.ts

```diff
@@ -163,6 +164,56 @@ export const useEventAction = () => {
     [_updateEventMutation],
   );
 
+  /**
+   * Calls mutation to edit multiple events
+   * @private
+   */
+  const _batchUpdateEventsMutation = useMutation({
+    mutationFn: requestBatchPutEvents,
+    onMutate: async ({ ids, data }) => {
+      // cancel ongoing queries
+      await queryClient.cancelQueries({ queryKey: RUNDOWN });
+
+      // Snapshot the previous value
+      const previousEvents = queryClient.getQueryData<GetRundownCached>(RUNDOWN);
+
+      if (previousEvents) {
+        const updatedEvents = previousEvents.rundown.map((event) => {
+          const isEventEdited = ids.includes(event.id);
+
+          if (isEventEdited && isOntimeEvent(event)) {
+            return {
+              ...event,
+              ...data,
+            };
+          }
+
+          return event;
+        });
+
+        queryClient.setQueryData(RUNDOWN, { rundown: updatedEvents, revision: -1 });
+      }
+
+      // Return a context with the previous and new events
+      return { previousEvents };
+    },
+    onSettled: async () => {
```

Comment:
I wonder if we should undo the mutation `onError`?

Reply:
Like this?
```ts
onError: (_error, _newEvent, context) => {
  queryClient.setQueryData(RUNDOWN, context?.previousEvents);
}
```